src/org/rascalmpl/parser/uptr/recovery/CaseInsensitiveLiteralMatcher.java

```diff
@@ -46,26 +46,16 @@ public CaseInsensitiveLiteralMatcher(int[][] ciLiteral) {
     }
 
     @Override
-    public MatchResult findMatch(int[] input, int startLocation, int maxLength) {
+    public int match(int[] input, int location) {
         int length = chars.length;
 
-        int limit = Math.min(startLocation + maxLength - length, input.length - length + 1);
-        for (int start=startLocation; start < limit; start++) {
-            boolean matches = true;
-            for (int i=0; i<length; i++) {
-                int inputChar = input[start+i];
-                if (inputChar != chars[i] && inputChar != altChars[i]) {
-                    matches = false;
-                    break;
-                }
-            }
-
-            if (matches) {
-                return new MatchResult(start, length);
+        for (int i=0; i<length; i++) {
```

Comment:
Shouldn't we guard that `length + location`  is smaller than `input.length`?

Reply:
This is done in the caller, but It will not hurt to do a little defensive programming here.